### PR TITLE
fix references for tags with a dot & non standard html tags

### LINKED ISSUE #1991: false positives for `no-unused-vars` in `gjs` files, when using `<Component as |something|>`
I am experiencing false positives for `no-unused-vars` in `.gjs` files

Example: 
```js
import UkCard from 'ember-uikit';

<template>
  <UkCard as |card|>
    <card.body>foo</card.body>
  </UkCard>
</template>
```
```
[lint:js]   /home[...]/ember-app/app/components/foo.gjs
[lint:js]   4:15  error  'card' is defined but never used  no-unused-vars
```
---
minimal reproduction: https://github.com/c0rydoras/gjs-linting-no-undef-reproduction
failing linting workflow: https://github.com/c0rydoras/gjs-linting-no-undef-reproduction/actions/runs/6798889824/job/18483951695

### PR DESCRIPTION
**Pull request by patricklx**
fixes #1991

### CODE REVIEW

File: lib/parsers/gjs-gts-parser.js [hunk 1]
```diff
@@ -260,6 +261,34 @@ function preprocessGlimmerTemplates(info, code) {
         n.loc.start = codeLines.offsetToPosition(tpl.templateRange[0]);
         n.loc.end = codeLines.offsetToPosition(tpl.templateRange[1]);
       }
+      // split up element node into sub nodes to be able to reference tag name
+      // parts <Foo.Bar /> -> nodes for `Foo` and `Bar`
+      if (n.type === 'ElementNode') {
```
Comment (NullVoxPopuli) [inline lib/parsers/gjs-gts-parser.js:266]: Shouldn't this be @glimmer/syntax responsibility?

Comment (patricklx) [reply lib/parsers/gjs-gts-parser.js:266]: i think it should, yes.
same for block params, there is no node for them

Comment (NullVoxPopuli) [reply lib/parsers/gjs-gts-parser.js:266]: Excellent, can you open an issue over on glimmer-vm?

Once we get that shipped, we can clean up here

**Review by NullVoxPopuli — APPROVED** [review]

**Comment by bmish** [comment]
@patricklx let's mention "gjs/gts parser" in the title of these PRs so users have the relevant context to understand which part of eslint-plugin-ember these PRs are affecting.

**Pull request merged**